elastic-search-support/src/test/java/uk/gov/hmcts/ccd/definition/store/elastic/ReindexServiceTest.java

```diff
@@ -143,6 +147,38 @@ void deletesNewIndexWhenReindexingFails() throws IOException {
         verify(ccdElasticClient, atLeast(2)).close();
     }
 
+    @Test
+    void shouldLogBeginReindexingMessageEvenWhenReindexFails() throws IOException {
+        Logger logger = (Logger) LoggerFactory.getLogger(ReindexService.class);
+        ListAppender<ILoggingEvent> appender = new ListAppender<>();
+        appender.start();
+        logger.addAppender(appender);
+
+        mockAliasResponse();
+
+        when(ccdElasticClient.countDocuments(caseTypeName)).thenReturn(50L);
+
+        doAnswer(invocation -> {
+            ReindexListener listener = invocation.getArgument(2);
+            listener.onFailure(new RuntimeException("reindexing failed"));
+            return null;
+        }).when(ccdElasticClient).reindexData(eq(caseTypeName), eq(incrementedCaseTypeName), any());
+
+        DefinitionImportedEvent event = new DefinitionImportedEvent(newArrayList(caseA), true, true);
+        assertThrows(RuntimeException.class,
+            () -> reindexService.asyncReindex(event, baseIndexName, caseA));
+
+        // assert begin reindexing log message is present
+        String combinedLogs = appender.list.stream()
```

Comment:
instead of checking the log, can't we check the count directly after the reindexing process? use countDocuments instead